CRAS cant be compiled

```
Compiling CRAS (i386)...
tests/cras_test_client.c: In function 'audio_debug_info':
tests/cras_test_client.c:585:10: error: format '%lu' expects argument of type 'long unsigned int', but argument 2 has type 'long long unsigned int' [-Werror=format=]
   printf("stream: %lu dev: %u\n",
          ^
tests/cras_test_client.c: In function 'run_aecdump':
tests/cras_test_client.c:677:10: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint64_t {aka long long unsigned int}' [-Werror=format=]
   printf("Dumping AEC info to %s, stream %lu, fd %d\n",
          ^
tests/cras_test_client.c: In function 'main':
tests/cras_test_client.c:1545:3: error: 'stream_id' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   run_aecdump(client, stream_id, 1);
   ^
tests/cras_test_client.c: At top level:
cc1: error: unrecognized command line option '-Wno-int-in-bool-context' [-Werror]
cc1: all warnings being treated as errors
/usr/lib/gcc/x86_64-linux-gnu/5/../../../i386-linux-gnu/crt1.o: In function `_start':
(.text+0x18): undefined reference to `main'
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...
chronos@localhost / $ god damnit...
```
whenever i try to install a chroot it always ends with this, ive deleted and reinstalled crouton (after assuming it was crouton) but it seems like more, anyone else ever had this issue?

and how did i cause this? i just used crouton to install xenial the normal way.

the command i used:
```
sudo sh -e ~/Downloads/crouton -t xorg,gtk-extra,cli-extra,core
```

Thanks, DennisLfromGA. Things here in Vegas were so very close.

chronos@localhost / $ sudo sh ~/Downloads/crouton -p '/media/removable/USB30a' -r buster -t kde_desktop,xiwi,chrome-dev -n buster3
Password: 
Invalid target "kde_desktop".
chronos@localhost / $ sudo sh ~/Downloads/crouton -p '/media/removable/USB30a' -r buster -t kde,xiwi,chrome-dev -n buster3
WARNING: USB booting is enabled; consider disabling it.
WARNING: Signed boot verification is disabled; consider enabling it.
You can use the following command: sudo crossystem dev_boot_usb=0 dev_boot_signed_only=1
Installing buster-amd64 chroot to /media/removable/USB30a/chroots/buster3
Downloading debootstrap 1.0.101...
################################################################################################################################################################# 100.0%################################################################################################################################################################# 100.0%
Patching debootstrap...
Downloading bootstrap files...
W: Cannot check Release signature; keyring file not available /usr/share/keyrings/debian-archive-keyring.gpg
I: Retrieving InRelease 
I: Retrieving Packages 
I: Validating Packages 
I: Resolving dependencies of required packages...
I: Resolving dependencies of base packages...
I: Checking component main on http://httpredir.debian.org/debian...
I: Retrieving libacl1 2.2.52-3+b1
I: Validating libacl1 2.2.52-3+b1
I: Retrieving adduser 3.117


and then:

he following packages were automatically installed and are no longer required:
  binutils binutils-common binutils-x86-64-linux-gnu gcc gcc-7 gcc-7-multilib gcc-multilib lib32asan4 lib32atomic1 lib32cilkrts5 lib32gcc-7-dev lib32gcc1 lib32gomp1
  lib32itm1 lib32mpx2 lib32quadmath0 lib32stdc++6 lib32ubsan0 libasan4 libasound2-dev:i386 libatomic1 libbinutils libc-dev-bin libc6-dev libc6-dev:i386 libc6-dev-i386
  libc6-dev-x32 libc6-i386 libc6-x32 libcc1-0 libcilkrts5 libdpkg-perl libgcc-7-dev libglib2.0-0 libitm1 liblsan0 libmpx2 libquadmath0 libspeex-dev:i386 libspeex1:i386
  libspeexdsp-dev:i386 libtsan0 libubsan0 libx32asan4 libx32atomic1 libx32cilkrts5 libx32gcc-7-dev libx32gcc1 libx32gomp1 libx32itm1 libx32quadmath0 libx32stdc++6
  libx32ubsan0 linux-libc-dev linux-libc-dev:i386 pkg-config
Use 'apt autoremove' to remove them.
Suggested packages:
  ed diffutils-doc
The following NEW packages will be installed:
  patch
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 123 kB of archives.
After this operation, 241 kB of additional disk space will be used.
Get:1 http://cdn-fastly.deb.debian.org/debian buster/main amd64 patch amd64 2.7.6-2 [123 kB]
Fetched 123 kB in 2s (65.8 kB/s)
Selecting previously unselected package patch.
(Reading database ... 15853 files and directories currently installed.)
Preparing to unpack .../patch_2.7.6-2_amd64.deb ...
Unpacking patch (2.7.6-2) ...
Setting up patch (2.7.6-2) ...
patch set to automatically installed.
Compiling CRAS (i386)...
tests/cras_test_client.c: In function 'audio_debug_info':
tests/cras_test_client.c:585:21: error: format '%lu' expects argument of type 'long unsigned int', but argument 2 has type 'long long unsigned int' [-Werror=format=]
   printf("stream: %lu dev: %u\n",
                   ~~^
                   %llu
          (unsigned long long)info->streams[i].stream_id,
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests/cras_test_client.c: In function 'run_aecdump':
tests/cras_test_client.c:677:44: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint64_t {aka long long unsigned int}' [-Werror=format=]
   printf("Dumping AEC info to %s, stream %lu, fd %d\n",
                                          ~~^
                                          %llu
tests/cras_test_client.c: In function 'main':
tests/cras_test_client.c:1545:3: error: 'stream_id' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   run_aecdump(client, stream_id, 1);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
/usr/lib/gcc/x86_64-linux-gnu/7/../../../i386-linux-gnu/Scrt1.o: In function `_start':
(.text+0x28): undefined reference to `main'
collect2: error: ld returned 1 exit status
Failed to complete chroot setup.
Unmounting /media/removable/USB30a/chroots/buster3...
chronos@localhost / $ 

But you've just given me an idea--I don't have to put 6 lb of meat in one giant ball.

Thanks, guys.

Mike.

--- a/targets/audio
+++ b/targets/audio
@@ -141,6 +141,14 @@ END
         # Drop SBC constants
         sed -e 's/SBC_[A-Z0-9_]*/0/g' -i tests/cras_test_client.c
 
+	# FIXME(crbug.com/864815) Remove when fixed upstream
+	if [ "$cras_arch" = "i386" ]; then
+	    sed -i -e 's/Dumping AEC info to %s, stream %lu/Dumping AEC info to %s, stream %llu/' \
+		tests/cras_test_client.c
+	fi
+	sed -i -e 's/cras_stream_id_t stream_id;/cras_stream_id_t stream_id = 0;/' \
+	    tests/cras_test_client.c
+
         # Directory to install CRAS library/binaries
         CRASLIBDIR="/usr/local$archextrapath/lib"
         CRASBINDIR="/usr/local$archextrapath/bin"
@@ -151,7 +159,7 @@ END
             convert_automake
 
             echo '
-                CFLAGS="$CFLAGS -Wno-int-in-bool-context -DCRAS_SOCKET_FILE_DIR=\"/var/run/cras\""
+                CFLAGS="$CFLAGS -DCRAS_SOCKET_FILE_DIR=\"/var/run/cras\""
 
                 buildlib libcras
 

